cpp/src/decisiontree/batched-levelalgo/builder_base.cuh

```diff
@@ -446,7 +452,9 @@ struct ClsTraits {
     // Pick the max of two
     size_t smemSize = std::max(smemSize1, smemSize2);
     dim3 grid(b.total_num_blocks, colBlks, 1);
-    CUDA_CHECK(cudaMemsetAsync(b.hist, 0, sizeof(int) * b.nHistBins, s));
+    int nHistBins = 0;
+    nHistBins = n_large_nodes_in_curr_batch * (1 + nbins) * colBlks * nclasses;
```

Comment:
0 initialising this variable is unnecessary.

The 1+ in (1+nbins) should disappear when you merge the objective function PR.

Reply:
ah, that was necessary in my initial prototyping version but missed it somehow.. changed it 👍🏾